### DISCUSSION TITLE
MLOps best practices

### DISCUSSION THREAD

**Discussion by KMayank29** [explosion/spaCy#10132] [Help: Best practices]
Hi Team spaCy,

My team has been working on a project where we clean data and fetch structured information out of it.
For this we train models (text classifier) with different configs using spaCy project and also data keeps on varying.
We feel it is necessary to keep track of model and data version as well as configs of the model.

We're looking for industry best practices in MLOps.
I reach out to spaCy team for their wise suggestions on MLOps for our project.

PS: We're considering using DVC by iterative.ai. 

Thanks and regards.  
K Mayank

**Comment by pmbaumgartner** [comment]
Hey @KMayank29 - It's hard to make generic recommendations on tools in this space, and I think best practices are still emerging. There are lots of tools out there depending on what you need. If you're new to MLOps, the best thing to do is to pick one and see if it meets your needs and if not, try out another tool on the next project.

While spaCy projects has DVC integration ([docs](https://spacy.io/usage/projects#dvc)), you can use any tool you want and automate the process through commands in your `project.yml` file.
